frontend/src/api/k8s/secrets.ts

```diff
@@ -26,7 +31,7 @@ export const assembleSecret = (
   const annotations: Record<string, string> = {};
 
   let stringData = data;
-  let name = `secret-${genRandomChars()}`;
+  let name = secretName || getGeneratedSecretName();
```

Comment:
Hmm... can you update the return block below -- it does a weird mix of this too -- seems odd logic.

Reply:
done